src/command/defaults/ClearCommand.php

```diff
@@ -110,65 +115,64 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
 			return true;
 		}
 
-		$cleared = 0;
-
-		//clear everything from the targets inventory
-		if($item === null){
-			$contents = array_merge($target->getInventory()->getContents(), $target->getArmorInventory()->getContents());
-			foreach($contents as $content){
-				$cleared += $content->getCount();
+		$clearedCount = 0;
+		if($targetItem === null){
+			// Clear all items from the inventories
+			$clearedCount += $this->countItems($inventories, null);
+			foreach($inventories as $inventory){
+				$inventory->clearAll();
 			}
-
-			$target->getInventory()->clearAll();
-			$target->getArmorInventory()->clearAll();
-			//TODO: should the cursor inv be cleared?
 		}else{
-			//clear the item from targets inventory irrelevant of the count
+			// Clear the item from target's inventory irrelevant of the count
 			if($maxCount === -1){
-				if(($slot = $target->getArmorInventory()->first($item)) !== -1){
-					$cleared++;
-					$target->getArmorInventory()->clear($slot);
-				}
-
-				foreach($target->getInventory()->all($item) as $index => $i){
-					$cleared += $i->getCount();
-					$target->getInventory()->clear($index);
+				$clearedCount += $this->countItems($inventories, $targetItem);
+				foreach($inventories as $inventory){
+					$inventory->remove($targetItem);
 				}
 			}else{
-				//clear only the given amount of that particular item from targets inventory
-				if(($slot = $target->getArmorInventory()->first($item)) !== -1){
-					$cleared++;
-					$maxCount--;
-					$target->getArmorInventory()->clear($slot);
-				}
+				// Clear the item from target's inventory up to maxCount
+				foreach($inventories as $inventory){
+					// Break if we've cleared enough items
+					if($maxCount === 0){
```

Comment:
This is an unnecessary check since below you are already checking before doing the next loop